src/lib/optimizer/strategy/dependent_group_by_reduction_rule.cpp

```diff
@@ -0,0 +1,132 @@
+        if (unique_columns.find(group_by_column) == unique_columns.end()) {
+          // Remove node if it is a column reference and references the correct stored table node. Further, decrement
+          // the aggregate's index which denotes the end of group-by expressions.
+          aggregate_node.node_expressions.erase(
+              std::remove_if(aggregate_node.node_expressions.begin(), aggregate_node.node_expressions.end(),
+                             [&, stored_table_node = stored_table_node](const auto expression) {
+                               const auto& column_expression =
+                                   std::dynamic_pointer_cast<LQPColumnExpression>(expression);
+                               if (!column_expression) return false;
+
+                               const auto& expression_stored_table_node =
+                                   std::dynamic_pointer_cast<const StoredTableNode>(
+                                       column_expression->column_reference.original_node());
+                               if (!expression_stored_table_node) return false;
+
+                               const auto column_id = column_expression->column_reference.original_column_id();
+                               if (stored_table_node == expression_stored_table_node && group_by_column == column_id) {
+                                 // Adjust the number of group by expressions.
+                                 --aggregate_node.aggregate_expressions_begin_idx;
+                                 group_by_list_changed = true;
+                                 return true;
+                               }
+                               return false;
+                             }),
+              aggregate_node.node_expressions.end());
+
+          // Add the ANY() aggregate to the list of aggregate columns.
+          const auto aggregate_any_expression = any_(lqp_column_({stored_table_node, group_by_column}));
+          aggregate_node.node_expressions.emplace_back(aggregate_any_expression);
+        }
+      }
+    }
+
+    // In case the initial query plan root returned the same columns in the same column order and was not a projection,
+    // it is likely that the result of the current aggregate was either the root itself or only operators followed that
+    // do not modify the column order (e.g., sort or limit). In this case, we need to restore the initial column order
+    // by adding a projection with the initial column_references since we changed the column order by moving columns
+    // from the group-by list to the aggregations.
+    if (group_by_list_changed && initial_aggregate_column_expressions == root_column_expressions &&
+        lqp->type != LQPNodeType::Projection) {
+      const auto projection_node = std::make_shared<ProjectionNode>(root_column_expressions);
+      lqp_insert_node(lqp, LQPInputSide::Left, projection_node);
+    }
```

Comment:
Why don't you compare the original and current column order in the root?

Reply:
Leftover from previous version. Changed to `aggregate_node.column_expressions() == root_column_expressions`.

Reply:
It is actually done by intent. We take the optimised aggregate's columns and if they have been equal to the root, there is probably no modifying operator following (that means limit, sort can follow, but when aggregate follows and is the root, we don't need to project).